atom/browser/native_window_mac.mm

```diff
@@ -322,6 +322,9 @@ void ViewDidMoveToSuperview(NSView* self, SEL _cmd) {
   }
 
   NSUInteger styleMask = NSWindowStyleMaskTitled;
+  if (!has_frame() && title_bar_style_ == NORMAL) {
```

Comment:
Only enable borderless window for `{ frame: false }`. Any other combo that involved non-default `titleBarStyle` will simply use the titled window.